packages/core/admin/ee/server/services/review-workflows/workflows/content-types.js

```diff
@@ -5,52 +5,67 @@ const { difference } = require('lodash/fp');
 const { getService } = require('../../../utils');
 const { WORKFLOW_MODEL_UID } = require('../../../constants/workflows');
 
-module.exports = ({ strapi }) => ({
-  /**
-   * Migrates entities stages. Used when a content type is assigned to a workflow.
-   * @param {*} options
-   * @param {Array<string>} options.srcContentTypes - The content types assigned to the previous workflow
-   * @param {Array<string>} options.destContentTypes - The content types assigned to the new workflow
-   * @param {Workflow.Stage} options.stageId - The new stage to assign the entities to
-   */
-  async migrate({ srcContentTypes = [], destContentTypes, stageId }) {
-    const { created, deleted } = diffContentTypes(srcContentTypes, destContentTypes);
-
-    await mapAsync(created, async (uid) => {
-      // If it was assigned to another workflow, transfer it from the previous workflow
-      const srcWorkflow = await getService('workflows').getAssignedWorkflow(uid);
-      if (srcWorkflow) {
-        // Updates all existing entities stages links to the new stage
-        await getService('stages').updateEntitiesStage(uid, { toStageId: stageId });
-        return this.transferContentType(srcWorkflow, uid);
-      }
-
-      // Create new stages links to the new stage
-      return getService('stages').updateEntitiesStage(uid, {
-        fromStageId: null,
-        toStageId: stageId,
+module.exports = ({ strapi }) => {
+  const contentManagerContentTypeService = strapi
+    .plugin('content-manager')
+    .service('content-types');
+
+  return {
+    /**
+     * Migrates entities stages. Used when a content type is assigned to a workflow.
+     * @param {*} options
+     * @param {Array<string>} options.srcContentTypes - The content types assigned to the previous workflow
+     * @param {Array<string>} options.destContentTypes - The content types assigned to the new workflow
+     * @param {Workflow.Stage} options.stageId - The new stage to assign the entities to
+     */
+    async migrate({ srcContentTypes = [], destContentTypes, stageId }) {
+      const { created, deleted } = diffContentTypes(srcContentTypes, destContentTypes);
+
+      await mapAsync(created, async (uid) => {
+        // If it was assigned to another workflow, transfer it from the previous workflow
+        const srcWorkflow = await getService('workflows').getAssignedWorkflow(uid);
+        if (srcWorkflow) {
+          // Updates all existing entities stages links to the new stage
+          await getService('stages').updateEntitiesStage(uid, { toStageId: stageId });
+          return this.transferContentType(srcWorkflow, uid);
+        }
+
+        await contentManagerContentTypeService.updateConfiguration(
+          { uid },
+          { options: { reviewWorkflows: true } }
+        );
```

Comment:
`options` is a new attribute that we are adding in the configuration If I recall correctly, but will this not completely override the previous options that were stored in the configuration?
We might use this for other features in the feature and if that is the behaviour now we may want to not override

Reply:
It's not deeply merged, it uses the `set(...)` method from Lodash. packages/core/content-manager/server/services/utils/store.js:33
And I fear that changing this set to use a `merge` method can change the behavior and create a breaking change.

Reply:
we can, but it won't solve this problem for others that might use the options object in future.

Reply:
I think all the stores in strapi use this "set", I recall us doing this for persisting tables and have also seen it for seat-enforcement. So maybe we could escalate this and propose something for this 🤔

Reply:
That would be a bigger topic that would not only affect RW though I imagine

Reply:
Yes, but on this subject, it's a store only used in configuration for content-manager.

Reply:
ok, what is your feeling about this then? do you think its worth doing the merge using a get here? Find something else?  leave it as it is...

Reply:
We can do the get and leave a comment, might be enough